### PR TITLE
JENKINS-70730 Don't remove id inside symbol

### PR DESCRIPTION
**Pull request by timja**
See [JENKINS-70730](https://issues.jenkins.io/browse/JENKINS-70730).

I can't see why we would need to remove existing ids in symbols, they're highly unlikely to be in an svg and this part of the code is before the caching is done.

Far better that we don't remove an id here than we do and break an svg completely....

### Testing done

Unit test + interactively tested with https://issues.jenkins.io/browse/JENKINS-70729 and the azure vm agents plugin.

### Proposed changelog entries

- JENKINS-70730 Don't remove id inside symbol

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] The Jira issue, if it exists, is well-described.
- [x] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)).
  - Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [x] There is automated testing or an explanation as to why this change has no tests.
- [x] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [x] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [x] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [x] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [x] For new APIs and extension points, there is a link to at least one consumer.

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`:

- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/7689"></a>

### CODE REVIEW

**Review by janfaracik — APPROVED** [review]

**Comment by daniel-beck** [comment]
Do these element IDs have different semantics than other IDs? Because short IDs like in the Jira issue look like they'd just invite ID collisions.

**Comment by timja** [comment]
> Do these element IDs have different semantics than other IDs? Because short IDs like in the Jira issue look like they'd just invite ID collisions.

They do not have different semantics you do have to make them unique (if they are different SVGs), ids are not that common and seem to be mostly used for gradients.

You could post-process an svg to make the ids unique if needed.

Here's a blog with a bit more info on it.
https://www.antonball.dev/blog/2020-06-15-svg-id-collision/

File: core/src/main/java/org/jenkins/ui/symbol/Symbol.java
Comment (mawinter69) [inline core/src/main/java/org/jenkins/ui/symbol/Symbol.java:97]: I think that can lead to a duplicated id for the svg itself.
Assume the svg has an id at the root `<svg id="symbolid" .../>`
Now I want to use that svg with 
`<l:icon sec="mysymbol" id="myid"/>`
The result will be 2 id attributes I think.

Comment (timja) [reply core/src/main/java/org/jenkins/ui/symbol/Symbol.java:97]: it could but why would the svg have a symbol at the root baked into it?

Otherwise to fix it this string replace hack needs replacing with a dom library that only operates on the svg tag and not children.

**Review by NotMyFault — APPROVED** [review]

**Comment by timja** [comment]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
